runtime/JavaScript/src/antlr4/tree/Trees.js

```diff
@@ -16,57 +17,57 @@ const Trees = {
      *  node payloads to get the text for the nodes.  Detect
      *  parse trees and extract data appropriately.
      */
-    toStringTree: function(tree, ruleNames, recog) {
+    toStringTree: function (tree, ruleNames, recog) {
         ruleNames = ruleNames || null;
         recog = recog || null;
-        if(recog!==null) {
+        if (recog !== null) {
             ruleNames = recog.ruleNames;
         }
         let s = Trees.getNodeText(tree, ruleNames);
         s = escapeWhitespace(s, false);
         const c = tree.getChildCount();
-        if(c===0) {
+        if (c === 0) {
             return s;
         }
         let res = "(" + s + ' ';
-        if(c>0) {
+        if (c > 0) {
             s = Trees.toStringTree(tree.getChild(0), ruleNames);
             res = res.concat(s);
         }
-        for(let i=1;i<c;i++) {
+        for (let i = 1; i < c; i++) {
             s = Trees.toStringTree(tree.getChild(i), ruleNames);
             res = res.concat(' ' + s);
         }
         res = res.concat(")");
         return res;
     },
 
-    getNodeText: function(t, ruleNames, recog) {
+    getNodeText: function (t, ruleNames, recog) {
         ruleNames = ruleNames || null;
         recog = recog || null;
-        if(recog!==null) {
+        if (recog !== null) {
             ruleNames = recog.ruleNames;
         }
-        if(ruleNames!==null) {
-            if (t instanceof RuleNode) {
+        if (ruleNames !== null) {
+            if (t instanceof RuleContext) {
```

Comment:
this change is incorrect, it might a future bug in the java runtime i.e. it so happens that currently only RuleContext implements RuleNode, but that could change

Reply:
We are not programming for the future. What you think might never come. What we do is to solve the current problems.

Reply:
Well there isn't any problem with using RuleNode, so why are you proposing this change ?